### PR TITLE
Newline in hostname

### PR DESCRIPTION
**Pull request by EiNSTeiN-**
When %0a is present in the hostname and `.normalize` is called on the URI object, the newline is unescaped, which is not desirable.

```
irb(main):001:0> require 'addressable/uri'
 => true
irb(main):002:0> url = Addressable::URI.parse('http://abc%0adef/')
 => #<Addressable::URI:0x3ffebdcbde2c URI:http://abc%0adef/>
irb(main):003:0> url
 => #<Addressable::URI:0x3ffebdcbde2c URI:http://abc%0adef/>
irb(main):004:0> url.to_s
 => "http://abc%0adef/"
irb(main):005:0> url.normalize.to_s
 => "http://abc\ndef/"
```

ping @sporkmonger

### CODE REVIEW

File: lib/addressable/uri.rb [hunk 1]
```diff
@@ -1094,7 +1098,7 @@ def host=(new_host)
 
       unreserved = CharacterClasses::UNRESERVED
       sub_delims = CharacterClasses::SUB_DELIMS
-      if @host != nil && (@host =~ /[<>{}\/\?\#\@"]/ ||
+      if @host != nil && (@host =~ /[<>{}\/\?\#\@"\r\n]/ ||
```
Comment (ptoomey3) [inline lib/addressable/uri.rb]: Maybe something like `[[:space:]]` would be more general purpose and allow us to catch more whitespace related values?

**Comment by ptoomey3** [comment]
Vaguely related: https://github.com/sporkmonger/addressable/pull/205

**Comment by EiNSTeiN-** [comment]
@sporkmonger, I updated this PR with your suggested changes. Sorry for the delay!

**Comment by sporkmonger** [comment]
All set. Sorry about the delay.

**Pull request merged**